app/helpers/generic_file_helper.rb

```diff
@@ -21,8 +21,9 @@ def render_download_link text = nil
     link_to (text || "Download"), download_path(@generic_file), { id: "file_download", target: "_new", data: { label: @generic_file.id } }
   end
 
-  def download_path gf
-    GenericFile.find(gf.id).doi_url || sufia.download_path(gf.id)
+  # sufia.download path is from ActionDispatch::Routing::PolymorphicRoutes#polymorphic_path
+  def download_path gf, options = {}
+    GenericFile.find(gf.id).doi_url || sufia.download_path(gf.id, options)
```

Comment:
Cool.  I think that makes sense. Is it safe to assume that the first argument is a GenericFile if that's the case?